### PR TITLE
I18n - part II

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: modules/@angular/compiler/src/i18n/extractor_merger.ts [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+      }
+      this._inIcu = true;
+    }
+
+    if (this._mode === _VisitorMode.Extract) {
+      html.visitAll(this, icu.cases, context);
+    }
+
+    if (this._mode === _VisitorMode.Merge) {
+      const cases = html.visitAll(this, icu.cases, context);
```
Comment (vicb) [inline modules/@angular/compiler/src/i18n/extractor_merger.ts]: factorize

File: modules/@angular/compiler/src/i18n/extractor_merger.ts [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+        this._reportError(
+            el, 'Could not mark an element as translatable inside a translatable section');
+      }
+
+      if (this._mode == _VisitorMode.Extract) {
+        // Descend into child nodes for extraction
+        html.visitAll(this, el.children);
+      }
+
+      if (this._mode == _VisitorMode.Merge) {
```
Comment (vicb) [inline modules/@angular/compiler/src/i18n/extractor_merger.ts]: add comment about icu

Comment (vicb) [inline modules/@angular/compiler/src/i18n/extractor_merger.ts]: adressed

**Comment by vicb** [comment]
@alxhub finally ready to be merged. Should not affect anyone. Thanks !

**Comment by alxhub** [comment]
Merged to master as 74b57df.

**Comment by vicb** [comment]
thanks alex !

**Pull request closed without merging**